docs/developer-guide/debugging.md

```diff
@@ -1,26 +1,130 @@
 
-!!! warning
-    We have had reports of problems with cheap clones of ST-Links not connecting.
+!!! note
+    There are also cheaper clones of the ST-LINK/V2 available. We have had fairly good luck with these, but have also run into some issues. These are a decent alternative, but we recommend the official version if you can afford it.
 
-## Add User to Dialout Group
+The following guide will show you how to get the in-circuit debugger running with either the Visual Studio Code or QtCreator IDE. Start with the steps in the [General Setup](#general-setup) section, then move on to either the [VS Code](#vs-code) or [QtCreator](#qtcreator) sections depending on your choice of IDE.
 
-!!! tip
-    You can see which groups you are in by running `groups $USER` on the command line.
+This guide assumes you are running Ubuntu 18.04, which is the currently supported development environment.
 
-First, make sure you are in the `dialout` group. If you are not in the `dialout` group, run:
+
+## General Setup
+
+Follow the guide in [Building and Flashing](/developer-guide/building-flashing) to install the compiler toolchain.
+
+Also make sure you have configured your computer as described in the [Serial Port Configuration](../user-guide/flight-controller-setup.md#serial-port-configuration) section of the user guide.
+
+### Connect debugger to flight controller
+
+The ST-LINK/V2 connects to the microcontroller using the Serial Wire Debug (SWD) interface. You will need to connect the `GND`, `NRST`, `SWDIO`, and `SWCLK` lines of the ST-LINK/V2 to your flight controller. On many F4 boards, these lines are pinned out through a 4-position JST SH connector, although that connector is not always populated. Refer to the documentation for your specific board for details.
+
+The official ST-LINK/V2 also needs a target voltage reference on pin 1 or 2, which for the F4 boards is 3.3V. However, there is no externally accessible 3.3V pinout on the F4 boards. An easy solution to this is to connect pin 19 (VDD 3.3V) of the ST-LINK/V2 to pin 1 or 2 of the ST-LINK/V2 (Target VCC) to provide the voltage reference. You will also need to power the board from another source, either through the USB port or over the servo rail.
+
+## VS Code
+
+You can install Visual Studio Code by downloading the latest version from [their website](https://code.visualstudio.com). Follow the steps below to configure debugging with the in-circuit debugger
```

Comment:
Do we want to put anywhere that you should open the root `firmware` folder?

Reply:
Also a good catch.